src/operator/contrib/deformable_convolution-inl.h

```diff
@@ -453,18 +454,10 @@ class DeformableConvolutionProp : public OperatorProperty {
                  std::vector<int> *out_type,
                  std::vector<int> *aux_type) const override {
     CHECK_GE(in_type->size(), 1U);
-    int dtype = (*in_type)[0];
-    CHECK_NE(dtype, -1) << "First input must have specified type";
-    for (size_t i = 0; i < in_type->size(); ++i) {
-      if ((*in_type)[i] == -1) {
-        (*in_type)[i] = dtype;
-      } else {
-        UNIFORM_TYPE_CHECK((*in_type)[i], dtype, ListArguments()[i]);
-      }
-    }
-    out_type->clear();
-    out_type->push_back(dtype);
```

Comment:
Wow, this would just discard the type inferred by the other operators...

Reply:
yep, many operators behave this way. probably more than what we have captured in the issue list :)

Reply:
Seems that the logic for type/shape inference in operators really needs to be looked at. I have another issue (and PR with a fix) coming today about how it is currently possible to not let the operator check their input and output attributes (and there should be also another how operators lie in their infertype/shape about whether they successfully inferred everything or not).